libglusterfs/src/unittest/mem_pool_unittest.c

```diff
@@ -102,7 +102,7 @@ helper_check_memory_headers(char *mem, xlator_t *xl, size_t size, uint32_t type)
     assert_int_equal(p->size, size);
     assert_true(p->xl == xl);
     assert_int_equal(p->header_magic, GF_MEM_HEADER_MAGIC);
-    assert_true(*(uint32_t *)(mem + sizeof(mem_header_t) + size) ==
+    assert_true(*(gf_mem_magic_t *)(mem + sizeof(mem_header_t) + size) ==
```

Comment:
Given that the trailer is written byte by byte in big-endian order now, this test will fail in little-endian architectures. We need to implement `__gf_mem_trailer_read()` here also.

Reply:
I have to admit, I don't fully grok why this unit-test isn't using the original code - seems to re-implement stuff from what it's testing?